Consolidate company name and details in sign up flow

TheDoubleK opened the pull request:
For this ticket: https://trello.com/c/79M7BHfG/44-sign-up-flow-consolidate-company-name-contact-details-pages-into-a-single-what-buyers-will-see-page

We used to have separate pages for company name and company contact details, now all those questions are on one page.
 
## The new page

![screen shot 2018-03-08 at 17 58 42](https://user-images.githubusercontent.com/6525554/37167645-64f57506-22fa-11e8-9233-bd88ccaa1573.png)

samuelhwilliams commented on tests/app/main/test_suppliers.py:
```diff
@@ -1639,8 +1624,9 @@ def test_should_not_allow_contact_details_without_phone_number(self):
+                'company_name': "My Company",
                 'contact_name': "Name",
                 'email_address': "name@email.com",
                 'phone_number': twentyone
```
We don't class an alpha string as 'invalid'?

We should, probably - but if we are going to leave it as-is, can we rename the test to just say that we mean it can't be over 20 chars.

TheDoubleK replied:
There's no way to reliably validate a phone number, people add spaces, dashes, plus symbols, etc.  And alphabetical can also work - phone keys have letters on them, and my bank used to advertise it's phone number as something like `0800-THEBANK`.

samuelhwilliams replied:
Yeah that's a fair point. I guess they can always change it later and they have to check/confirm it.

TheDoubleK commented on app/main/views/suppliers.py:
```diff
@@ -517,80 +516,48 @@ def submit_duns_number():
+                "suppliercreate.fail: duns:{duns} company_name:{company_name} {form_errors}",
+                extra={
+                    'duns': session.get('duns_number'),
+                    'company_name': session.get('company_name'),
```
Actually, I think maybe this should not have `company_name` and only DUNS number, as that's the only thing that will have been set in the session up to now?

Company name, etc. only get set if the form validates.  If it hasn't validated then none of these things will have values in the session.

We could use the values from the form rather than the values from the session to populate this?  But to be honest I'm not sure there's much value in logging validation errors at all.

samuelhwilliams replied:
I think all of the other instances of this logging grab the data from the form, so getting it from the session at all is kinda weird and probably wrong.

Although as you say, it's questionable the value it has...

TheDoubleK commented on app/main/views/suppliers.py:
```diff
@@ -517,80 +516,48 @@ def submit_duns_number():
-def company_name():
-    form = CompanyNameForm()
+@main.route('/company-details', methods=['GET', 'POST'])
+def company_details():
```
It might be nice to do this.  I'll check with Ash how much of a headache it will cause him.

CrystalPea replied:
Awesome, please let me know, it would be great and I am happy to help :).

TheDoubleK commented:
Ash says the new URLs are fine to do, so I've told him what they are and updated the code here.  Definitely an improvement I think.

@samuelhwilliams also fixed the test name and log output.

CrystalPea commented:
@TheDoubleK wooo well done <3
